test/Conversion/SCFToCalyx/convert_controlflow.mlir

```diff
@@ -0,0 +1,130 @@
+// CHECK-NEXT:             calyx.seq  {
+// CHECK-NEXT:               calyx.seq  {
+// CHECK-NEXT:               }
+// CHECK-NEXT:               calyx.seq  {
+// CHECK-NEXT:                 calyx.enable @bb1_to_bb3 {compiledGroups = []}
+// CHECK-NEXT:               }
+// CHECK-NEXT:               calyx.enable @ret_assign_0 {compiledGroups = []}
+// CHECK-NEXT:             }
+// CHECK-NEXT:           } else  {
+// CHECK-NEXT:             calyx.seq  {
+// CHECK-NEXT:               calyx.seq  {
+// CHECK-NEXT:               }
+// CHECK-NEXT:               calyx.seq  {
+// CHECK-NEXT:                 calyx.enable @bb2_to_bb3 {compiledGroups = []}
+// CHECK-NEXT:               }
+// CHECK-NEXT:               calyx.enable @ret_assign_0 {compiledGroups = []}
+// CHECK-NEXT:             }
+// CHECK-NEXT:           }
+// CHECK-NEXT:         }
+// CHECK-NEXT:       }
+// CHECK-NEXT:     }
+// CHECK-NEXT:   }
+// CHECK-NEXT: }
+module {
+  func @main(%arg0 : i32, %arg1 : i32) -> i32 {
+    %0 = cmpi slt, %arg0, %arg1 : i32
+    cond_br %0, ^bb1, ^bb2
+  ^bb1:
+    br ^bb3(%arg0 : i32)
+  ^bb2:
+    br ^bb3(%arg1 : i32)
+  ^bb3(%1 : i32):
+    return %1 : i32
+  }
+}
+
+// -----
+
+// CHECK:      module  {
+// CHECK-NEXT:   calyx.program "main"  {
+// CHECK-NEXT:     calyx.component @main(%in0: i32, %in1: i32, %in2: i32, %clk: i1 {clk}, %reset: i1 {reset}, %go: i1 {go}) -> (%out0: i32, %done: i1 {done}) {
+// CHECK-NEXT:       %true = hw.constant true
+// CHECK-NEXT:       %std_slt_0.left, %std_slt_0.right, %std_slt_0.out = calyx.std_slt "std_slt_0" : i32, i32, i1
+// CHECK-NEXT:       %while_0_arg0_reg.in, %while_0_arg0_reg.write_en, %while_0_arg0_reg.clk, %while_0_arg0_reg.reset, %while_0_arg0_reg.out, %while_0_arg0_reg.done = calyx.register "while_0_arg0_reg" : i32, i1, i1, i1, i32, i1
+// CHECK-NEXT:       %ret_arg0_reg.in, %ret_arg0_reg.write_en, %ret_arg0_reg.clk, %ret_arg0_reg.reset, %ret_arg0_reg.out, %ret_arg0_reg.done = calyx.register "ret_arg0_reg" : i32, i1, i1, i1, i32, i1
+// CHECK-NEXT:       calyx.wires  {
+// CHECK-NEXT:         calyx.assign %out0 = %ret_arg0_reg.out : i32
+// CHECK-NEXT:         calyx.group @assign_while_0_init  {
+// CHECK-NEXT:           calyx.assign %while_0_arg0_reg.in = %in0 : i32
+// CHECK-NEXT:           calyx.assign %while_0_arg0_reg.write_en = %true : i1
+// CHECK-NEXT:           calyx.group_done %while_0_arg0_reg.done : i1
+// CHECK-NEXT:         }
+// CHECK-NEXT:         calyx.comb_group @bb0_0  {
+// CHECK-NEXT:           calyx.assign %std_slt_0.left = %while_0_arg0_reg.out : i32
+// CHECK-NEXT:           calyx.assign %std_slt_0.right = %in1 : i32
+// CHECK-NEXT:         }
+// CHECK-NEXT:         calyx.group @assign_while_0_latch  {
+// CHECK-NEXT:           calyx.assign %while_0_arg0_reg.in = %while_0_arg0_reg.out : i32
+// CHECK-NEXT:           calyx.assign %while_0_arg0_reg.write_en = %true : i1
+// CHECK-NEXT:           calyx.group_done %while_0_arg0_reg.done : i1
```

Comment:
How painful would it be to turn this into something meaningful, e.g. increment the value, instead of `reg.in = reg.out`.

Reply:
It's a bit difficult to make more meaningful tests until combinational group inlining has been implemented. At this point, only register outputs and component inputs can be used...

Reply:
Ah ok makes sense :) . Maybe leave a TODO then to update this.